crates/catalog/rest/src/catalog.rs

```diff
@@ -963,6 +993,76 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn test_get_default_headers() {
+        let server = Server::new_async().await;
+        let mut props = HashMap::new();
+        props.insert("credential".to_string(), "client1:secret1".to_string());
+
+        let config = RestCatalogConfig::builder()
+            .uri(server.url())
+            .props(props)
+            .build();
+        let headers: HeaderMap = config.get_default_headers().unwrap();
+
+        let expected_headers = HeaderMap::from_iter([
+            (
+                header::CONTENT_TYPE,
+                HeaderValue::from_static("application/json"),
+            ),
+            (
+                HeaderName::from_static("x-client-version"),
+                HeaderValue::from_static(ICEBERG_REST_SPEC_VERSION),
+            ),
+            (
+                header::USER_AGENT,
+                HeaderValue::from_str(&format!("iceberg-rs/{}", CARGO_PKG_VERSION)).unwrap(),
+            ),
+        ]);
+        assert_eq!(headers, expected_headers);
+    }
+
+    #[tokio::test]
+    async fn test_get_default_headers_with_custom_headers() {
+        let server = Server::new_async().await;
+        let mut props = HashMap::new();
+        props.insert("credential".to_string(), "client1:secret1".to_string());
+        props.insert(
+            "header.content-type".to_string(),
```

Comment:
For REST we have the `rest.` prefix for REST-specific config, see: https://py.iceberg.apache.org/configuration/

We might want to consider it here as well, WDYT?

Reply:
@Fokko Can we address that using a separate PR? Because this PR is to resolve https://github.com/apache/iceberg-rust/issues/292. If you can help file an issue (to support `rest.`), I will be happy to work on it.

Reply:
Checked the Java and Python's implementation. Both are without `rest.`  for headers. I'd be nice to have a `rest.` prefix. It doesn't seem a blocker to me though.

pyIceberg doc from https://github.com/apache/iceberg-python/pull/467:
```yaml
catalog:
  default:
    uri: http://rest-catalog/ws/
    credential: t-1234:secret
    header.content-type: application/vnd.api+json
```

Java config example
```
spark.sql.catalog.my-catalog.header.X-Catalog-Name=default 
```